gen/field.go

```diff
@@ -147,23 +147,24 @@ func generateTags(f *compile.FieldSpec) (string, error) {
 }
 
 func compileJSONTag(f *compile.FieldSpec, name string, opts ...string) *structtag.Tag {
-	// We want to add omitempty if the field is an optional struct or
-	// primitive to reduce "null" noise. We won't add omitempty for
-	// optional collections because omitempty doesn't differentiate
-	// between nil and empty collections.
-
 	t := &structtag.Tag{
 		Key:     jsonTagKey,
 		Name:    name,
 		Options: opts,
 	}
 
+	// If the field name is "-" then it means omit, add no tags
 	if name == "-" {
-		// If the field name is "-" then it means omit, add no tags
 		return t
 	}
 
-	if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required && !t.HasOption("omitempty") {
+	// We want to add the "omitempty" json tag if the field is an 'optional" to
```

Comment:
Nit.

Either:

    is optional

Or

    is an "optional".

Also, mismatch single and double quote.

Reply:
Done.